### PR TITLE
Document the application's design

### PR DESCRIPTION
**Pull request by hosamaly**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,28 +2,92 @@
...
+by Amazon's
+[Kindle Publishing for Periodicals (KPP)](https://kindlepublishing.amazon.com/gp/vendor/kindlepubs/kpp/kpp-home)
+in order to be published on the
+[Kindle Newsstand](https://www.amazon.co.uk/Magazines-Journals-Kindle/b?ie=UTF8&node=341690031).
 
-# Prerequisites for development 
+## Design
+The application has been designed with extensibility in mind. It should be possible to use its
+core, unmodified, to support the generation of different newsletters. The Guardian's Kindle Edition
+is is one example that is baked into this repository.
```
Comment (regiskuckaertz) [inline README.md]: ~~is~~

File: README.md [hunk 2]
```diff
@@ -2,28 +2,92 @@
...
-```sh
-sbt -Dcontent-api.key="<your-key>" [yourCommand]
-```
+### The Core
+The main entrypoint is `KindleGenerator`. It accepts an `ArticlesProvider` that is responsible
+for providing the HTML articles to be published in the edition. An example implementation is the
+`DailyWeatherForecastProvider`.
 
+With the provided articles, the `KindleGenerator` proceeds to (optionally) download the articles'
+main images. It accepts a `Downloader`, such as `OkHttpSttpDownloader`, give the caller full
```
Comment (regiskuckaertz) [inline README.md]: `s/give/giving/`

**Comment by LATaylor-guardian** [comment]
Nice work @hosamaly 👍

**Comment by LATaylor-guardian** [comment]
@hosamaly Why did you close this?

**Comment by hosamaly** [comment]
@LATaylor-guardian: It wasn't actually closed but the branch from which it was created had been deleted. I set the base branch to master and it's showing up correctly now.

**Pull request closed without merging**